events/events_network.c

```diff
@@ -136,6 +136,15 @@ events_network_register(int (*func)(void *), void * cookie, int s, int op)
 		goto err0;
 
 	/*
+	 * There is a new socket/operation pair:
+	 * Rebuild the fd list on the next call to events_network_select().
+	 */
+	if (fds != NULL) {
```

Comment:
Not good.  In the common case that a callback reads/writes data and then immediately re-registers itself to read/write more, this will result in the fd list being regenerated after every network I/O.  Worse, if there are many readable/writable sockets, the lowest-numbered one will always get the chance to do something and the higher-numbered sockets will get starved.